llvm/test/TableGen/listflatten-error.td

```diff
@@ -0,0 +1,6 @@
+// RUN: not llvm-tblgen %s 2>&1 | FileCheck %s  -DFILE=%s
+
+// CHECK: [[FILE]]:[[@LINE+2]]:33: error: expected argument of type list of list in !listflatten operator
+class Flatten<list<int> A> {
+    list<int> F = !listflatten(A);

```

Comment:
Test with some non-list types too. For the case of an already flat list, should it really be an error or a no-op?

Reply:
Resolving this. Added test for non-list type and any list is now accepted by this operator.